src/backend/gl/src/window/glutin.rs

```diff
@@ -154,55 +306,424 @@ impl Device {
+                            width: config.extent.width,
+                            height: config.extent.height,
+                        };
+
+                        ContextMaker::new_wayland(
+                            ctxt,
+                            &full_wci,
+                        ).unwrap()
+                    } else {
+                        unreachable!()
+                    }
+                }
+                Contexts::SurfaceContext(_) => unreachable!(),
+            };
+
+            #[cfg(windows)]
+            let (window, events_loop, context) = match *ctxts {
+                Contexts::InstanceContext(ref ic) => {
+                    let ic = Wstarc::upgrade(ic).unwrap();
+                    let ctxt = &ic.context;
+
+                    ContextMaker::new_windows(ctxt, &s.wci).unwrap()
+                }
+                Contexts::SurfaceContext(_) => unreachable!(),
+            };
+
+            let events_loop = Starc::new(events_loop);
+
+            let surface_context = Arc::new(SurfaceContext {
+                window,
+                events_loop,
+                context,
+            });
+
+            // There is a possiblity that the function pointers from one context
+            // aren't valid with the other. We ere on the side of caution and
+            // just open a new device (and therefore a new Share).
+            // NOTE: We discard the name and VAO made by new/to_device cause we
+            // don't need them.
+
+            let (_, pd) = PhysicalDevice::new(
+                |s| surface_context.context.get_proc_address(s) as *const _,
+                Contexts::SurfaceContext(Arc::downgrade(&surface_context)),
+            );
+            let (_, device) = pd.to_device().unwrap();
+
+            *surface = Surface::Swapchained(Arc::new(SurfaceSwapchained {
+                surface_context,
+                device,
+            }));
+        }
+
+        let swapchain = if let Surface::Swapchained(surface) = surface {
+            surface.surface_context.context.resize(glutin::dpi::PhysicalSize {
+                width: config.extent.width as _,
+                height: config.extent.height as _,
+            });
+
+            let mut images = vec![];
+            for _ in 0..config.image_count {
+                let mut fbo = 0;
+                unsafe {
+                    let gl = &surface.device.share.context;
+                    gl.GenFramebuffers(1, &mut fbo);
+                }
+
+                use hal::Device;
+                let kind = image::Kind::D2(config.extent.width, config.extent.height, 1, 1);
+                let uimage = surface.device.create_image(
+                    kind,
+                    1,
+                    f::Rgba8Srgb::SELF,
+                    image::Tiling::Optimal, // ignored
+                    image::Usage::SAMPLED, // FIXME: Hello reviewer! Is this the correct one? I'm trying to get a texture, not a surface, and this appears to be the way.
```

Comment:
Is this the correct one? I'm trying to get a texture, not a surface, and this appears to be the way.